lib/Dialect/Handshake/HandshakeOps.cpp

```diff
@@ -1428,28 +1428,37 @@ void JoinOp::build(OpBuilder &builder, OperationState &result,
   result.types.push_back(type);
 
   result.addOperands(operands);
-  sost::addAttributes(result, operands.size(), type);
+  if (llvm::all_of(operands.getTypes(),
+                   [](Type t) { return t.isa<NoneType>(); }))
+    result.addAttribute("control", BoolAttr::get(result.getContext(), true));
```

Comment:
Is this not always a control-only operator? Regardless of the input type, the will only ever care about the handshake signals, and emit a `none`-typed signal.

Reply:
The FIRRTL lowering uses this attribute to determine if it has to care about naming the module with the input types or not. Thus I thought that the presence of this attribute indicates that an operation only works with `none` types. 

We can change this, but then the lowering to FIRRTL requires some additional changes.

Reply:
I see. Yea, this attribute seems somewhat arbitrary - Ideally, this information should be available by anyways inspecting the in/out types of an op. In this case, LGTM - but we should probably change it for the `HandshakeToHW` lowering (and get rid of the attribute).

Reply:
NVM, I confused something. 
I'll change it